ament_index_python/ament_index_python/packages.py

```diff
@@ -44,7 +45,11 @@ def get_package_prefix(package_name):
     :param str package_name: name of the package to locate
     :returns: installation prefix of the package
     :raises: :exc:`PackageNotFoundError` if the package is not found
+    :raises: :exc:`ValueError` if the package name is invalid
     """
+    if re.fullmatch('[a-z][a-z0-9_]+', package_name, re.ASCII) is None:
```

Comment:
I guess it's not convenient to use an already existing check for this, e.g. from `ament_package` or something? (could very well be the case that it's not worth the effort)

If not, then I think we should have a back reference to REP 144 (or 127, which ever is appropriate) here as a comment. Otherwise this regex seems to have been plucked out of no where, no matter how obvious is it seems to us, it might be confusing in the future.

Reply:
Closest thing I could find for an existing check is [in `catkin_pkg`](https://github.com/ros-infrastructure/catkin_pkg/blob/ad07ab2570006a72994657ab3f3267a9e84469aa/src/catkin_pkg/package.py#L231-L238). The match differs a little there as it includes the dashes and uppercase letters from REP-127's ["recommended exemptions"](https://ros.org/reps/rep-0127.html#name). It's not really in a reusable form there.

Dashes should be added to this match too to allow for the full range of package names that might show up from a manifest.

Reply:
Changed to match REP-127 and `catkin_pkg`. Added a comment explaining source